### PR TITLE
Added Pinterest Destination

### PR DESCRIPTION
**Pull request by Shivam4415**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by ameypv-rudder — APPROVED** [review]

File: destinations/pinterest-tag.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+---
+description: Step-by-step guide to send your event data from RudderStack to Pinterest.
+---
+
+# Pinterest Tag
+
+The [Pinterest Tag](https://ads.pinterest.com/advertiser/549762398656/conversions/tag/) is a piece of code that, when added to your website, allows Pinterest to track your visitors as well as their actions. It gives you a detailed conversion report related to any and all customer activities like signups or buying a product.
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:7]: The URL here is redirecting to `https://ads.pinterest.com/`, so let's that URL?

File: destinations/pinterest-tag.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+RudderStack supports Pinterest as a destination to which you can send your customer events in real-time, through Pinterest's `pintrk` conversion tag.
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:9]: Should we add a git link pointing to our implementation? Similar to others

File: destinations/pinterest-tag.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+![Pinterest Tag Connection Settings in RudderStack](../.gitbook/assets/pinterest_tag.png)
+
+## Enable Enhanced Match
+
+Rudderstack supports Pinterest Enhanced Match in the following two scenarios:
+
+* Where a user is identified everytime they visit your site.
+* When a user visits your site anonymously but is identified at some later point by making an `identify()` call.
+
+To support Pinterest Enhanced Match, go to the Pinterest Tag destination settings in the Rudderstack dashboard and turn on the **Enable Enhanced Match on Page Load** option.
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:46]:
```suggestion
To support Pinterest Enhanced Match, go to the Pinterest Tag destination settings in the RudderStack dashboard and turn on the **Enable Enhanced Match on Page Load** option. 
```

File: destinations/pinterest-tag.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+Rudderstack supports Pinterest Enhanced Match in the following two scenarios:
+
+* Where a user is identified everytime they visit your site.
+* When a user visits your site anonymously but is identified at some later point by making an `identify()` call.
+
+To support Pinterest Enhanced Match, go to the Pinterest Tag destination settings in the Rudderstack dashboard and turn on the **Enable Enhanced Match on Page Load** option. 
+
+### Enable Enhanced Match on Page Load
+
+When turned `on` this configuration will attach the hashed email address on the initial page load. Now any call made to Pinterest will be an Enhanced Match.
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:50]:
```suggestion
When turned `ON` this configuration will attach the hashed email address on the initial page load. Now any call made to Pinterest will be an Enhanced Match.
```

Does it look better? @ameypv-rudder

File: destinations/pinterest-tag.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+When turned `off` all visits made to your site will be an anonymous. But you can identify any user at later point by making our `identify()` calls. 
+
+If you use Rudderstack's `identify()` method to enable Pinterest’s Enhanced Match, you can only collect this information for successive events. Pinterest does not update the values for the past events retroactively.
+
+## Identify
+
+When you make an `identify()` call with the user’s email address and traits, RudderStack triggers a Pinterest `set()` method. This saves the identification parameters so they can be sent with the next events, so it’s important to set the values as early as possible.
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:58]:
```suggestion
When you make an `identify()` call with the user’s email address and traits, RudderStack triggers a Pinterest `set()` method. This saves the identification parameters to be sent with the next events, so it’s important to set the values as early as possible. 
```

File: destinations/pinterest-tag.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+A sample `track` call looks like the following:
+
+```javascript
+rudderanalytics.track("Track me");
+```
+
+RudderStack's SDK will send the `track` event name and any properties as custom properties to Pinterest.
+
+## eCommerce
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:86]:
```suggestion
## E-Commerce
```

File: destinations/pinterest-tag.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+RudderStack supports eCommerce conversion tracking for Pinterest. Use the [RudderStack eCommerce Specification](https://docs.rudderstack.com/rudderstack-api-spec/rudderstack-ecommerce-events-specification) for sending the events while instrumenting your site with the RudderStack SDK.
+
+Below are some examples of the `track` event names that are passed to the Pinterest event:
+
+| RudderStack event name | Pinterest event name |
+| :--- | :--- |
+| Order Completed | `Checkout` |
+| Order Completed | `Checkout` |
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:95]: Repeated line. Please remove one

File: destinations/pinterest-tag.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+If you wish to pass any extra properties other than what we support, you can use our **Custom Properties** section while configuring your Pinterest destination in RudderStack. 
+
+### Custom Properties
+
+If you wish to send your defined properties to Pinterest, you can do so by using the **Custom Properties** section. Just add those properties in the given field and you are good to go. 
+
+Here is an example of how it all works:
+
+```javascript
+rudderanalytics.track('Event', {customProperty: { customValue: 2 }, someRandomMailId: 'user@gmail.com'})
```
Comment (arnab-p) [inline destinations/pinterest-tag.md:161]: Please format this snippet

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: destinations/pinterest-tag.md (hunk 1 above)
Comment (ameypv-rudder) [reply destinations/pinterest-tag.md:7]: makes sense.

File: destinations/pinterest-tag.md (hunk 4 above)
Comment (ameypv-rudder) [reply destinations/pinterest-tag.md:50]: Tweaked this entirely.

File: destinations/pinterest-tag.md (hunk 2 above)
Comment (ameypv-rudder) [reply destinations/pinterest-tag.md:9]: Absolutely. Please share the link to our implementation.

Comment (Shivam4415) [reply destinations/pinterest-tag.md:9]: https://github.com/rudderlabs/rudder-sdk-js/tree/production/integrations/PinterestTag

**Pull request merged**